nft/src/lib.rs

```diff
@@ -171,10 +191,15 @@ impl<T: Trait> Module<T> {
 	) -> Result<T::TokenId, DispatchError> {
 		NextTokenId::<T>::try_mutate(class_id, |id| -> Result<T::TokenId, DispatchError> {
 			let token_id = *id;
-			*id = id.checked_add(&One::one()).ok_or(Error::<T>::NoAvailableTokenId)?;
-
 			Classes::<T>::try_mutate(class_id, |class_info| -> DispatchResult {
 				let info = class_info.as_mut().ok_or(Error::<T>::ClassNotFound)?;
+
+				ensure!(
+					info.mint_mode == MintMode::Public || (info.mint_mode == MintMode::Private && info.owner == *owner),
```

Comment:
This is probably extraneous condition check? Can it be replaced with
`info.mint_mode == MintMode::Public || info.owner == *owner`?

Reply:
That's a good question.
I prefer the concept to write stateless checks. In this case, enumeration state.
For example, if we added the Invalid option, then in the existing version it would not work

Reply:
Ok, makes sense, thanks!